apps/autoscheduler/CMakeLists.txt

```diff
@@ -0,0 +1,169 @@
+# weights
+file(GLOB WF baseline.weights)
+get_filename_component(BASENAME_WF ${WF} NAME_WE)
+set(WF_CPP ${BASENAME_WF}.cpp)
+
+add_custom_command(OUTPUT ${WF_CPP}
+                   DEPENDS ${WF} binary2cpp
+                   COMMAND binary2cpp ${BASENAME_WF}_weights < ${WF} > ${WF_CPP}
+                   COMMENT "${WF} -> ${WF_CPP}"
+                   VERBATIM)
+
+# =======================================================
+
+# retrain_cost_model
+add_executable(retrain_cost_model_process
+               ASLog.cpp
+               DefaultCostModel.cpp
+               Weights.cpp
+               retrain_cost_model.cpp
+               ${WF_CPP})
+set_target_properties(retrain_cost_model_process
+                      PROPERTIES OUTPUT_NAME retrain_cost_model)
+
+set(COST_MODEL_GENERATOR_FILES cost_model_generator.cpp)
+halide_generator(cost_model.generator SRCS ${COST_MODEL_GENERATOR_FILES})
+halide_generator(train_cost_model.generator SRCS ${COST_MODEL_GENERATOR_FILES})
+halide_library_from_generator(cost_model
+                              FUNCTION_NAME
+                              cost_model
+                              GENERATOR
+                              cost_model.generator)
+halide_library_from_generator(train_cost_model
+                              FUNCTION_NAME
+                              train_cost_model
+                              GENERATOR
+                              train_cost_model.generator)
+target_include_directories(retrain_cost_model_process
+                           PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../support)
+target_link_libraries(retrain_cost_model_process
+                      PRIVATE cost_model train_cost_model)
+add_dependencies(retrain_cost_model_process cost_model train_cost_model)
+
+# =======================================================
+
+# libauto_schedule
+add_library(auto_schedule
+            SHARED
+            ASLog.cpp
+            AutoSchedule.cpp
+            DefaultCostModel.cpp
+            FunctionDAG.cpp
+            Weights.cpp
+            ${WF_CPP})
+
+if(HALIDE_LIBRARY_TYPE STREQUAL STATIC)
+  target_link_libraries(auto_schedule PRIVATE cost_model train_cost_model)
+else()
+  target_link_libraries(auto_schedule
+                        PRIVATE cost_model train_cost_model Halide)
+endif()
+
+if(NOT MSVC)
+  set_target_properties(auto_schedule PROPERTIES LINK_FLAGS "-rdynamic")
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL Clang)
```

Comment:
I think `-undefined dynamic_lookup` is specific to OSX, not Clang, which would mean this would break for someone using Clang on Linux